### ISSUE TITLE
new website

### ISSUE THREAD

**Issue by gr2m** [janl/offlinefirst.org#119]
@kidfribble and I met this weekend and discussed how to structure the new offlinefirst.org website. This is a long time coming, and I still owe you all a blog post based on the discussions we had in our first two Offline Camps (sorry @terichadbourne! Did not forget!). Our main goal is to make offlinefirst.org more collaborative, more owned by its community. We want to make it simpler to contribute to. At the same time, we want to keep it small and use it more like hub that references all the great projects from within and outside our community.

I will update this issue’s description based on @kidfribble and everyone else’s feedback

### Structure

- welcome
  - process
  - resources
  - news
  - events
  - contribute

#### Welcome

Shorter introduction followed by content excerpts from the other sections, to give it more of an page overview feel

#### Process

Offline First is an approach / design process. You start with the most constraint environment first etc etc. We were thinking to turn our process and best practices into a little guide, something like Dockyard’s [Design Sprints guide](https://dockyard.com/design-sprints/). This will be well written, concise and always up-to-date.

#### Resources

Basically an [awesome offline-first list](https://github.com/pazguille/offline-first). With sub categories like tools, technologies, tutorials, design patterns, case studies, bussines cases, etc. 

#### news

In addition to [what we have today](http://offlinefirst.org/news/) we suggest to post links to interesting news all over the internet as they occur. We will of course keep our newsletter and keep links to past issues.

#### contribute

Here is where we want to describe how people can contribute to the Open Source movement. Not only by contributing to the website, but by giving talks themselves, organising meetups and events, sharing their Open Source code, etc.

**Comment by terichadbourne** [comment]
Thanks for sharing @gr2m! I'd love to be involved w/ you and @kidfribble as this gets further fleshed out. Some initial questions...

- For the **Resources** page, does that have to be a static list where each resources sits under one category, and requires more and more scrolling as time goes on, or can it be a faceted search engine? Ideally we would have some database of all the resources we've found, tagged with format types (case study, event, video, podcast, interview, blog, article, etc.) and relevant content tags (UX, service workers, pouchdb, etc), and the user could control what subset of this they saw on the website.  

- Re **News** and **Resources** - I would hope that all the awesome things @renrutnnej has found for the newsletter would also be included in the resources database and pull into listings of videos, case studies, tutorials, etc. there. In my mind, any suggestions people have for resources would go into one central spot to be added to both the site and the newsletter. Anything I could find (slowly) by opening each old newsletter and scrolling through I could also find instead by doing the faceted search in the resources database. (I was longing for this when I scoured the newsletter for podcasts recently - the tags in the newsletter were super helpful, but the process was time-consuming.)   We could definitely save overall time by making it easier for Jenn & I to share resources we're hunting down instead of both doing our own hunting (currently her for newsletter and me for camp Twitter and YouTube).

- For quick reference, here's the very basic **Resources** page I built on Medium w/ links out to YouTube, portions of Offlinefirst.org, details on how to get involved, etc. - this is both "how to learn more" and "how to contribute": https://medium.com/offline-camp/offline-first-resources-2acc5836e9d4#.4ktme1t32 (not suggesting this is what the new resources or contributions page will look like, just want to ensure everything included here is also very easy to access from the new site.

- You included **Events** in your bulleted list but not your detailed page listing underneath. Do you still see that being its own page? If so, some  issues I'd like to account for there include: 
(1) Being able to list multiple sessions per event, with separate URLs for each session plus one for the overall event.
(2) Ability to differentiate btw event formats (conference, meetup, etc.)
(3) Ability to include speaker names and link to their Twitter accounts or whatnot
^^^ I can force this all to happen manually if needed, just suggesting we could build in styling for the options we're most likely to need
(4) Standard approach for dealing with past events and recorded sessions (right now we have a link to our YouTube channel
(5) Is there a case for storing events as resources as we do the others (see my database suggestion below). If so, would it be helpful to have records to which we later add links to slide decks and videos of completed presos?

- Note that the website currently doesn't have a good place for **podcast** episodes, which I'm instead listing here: https://medium.com/offline-camp/offline-first-podcasts-d0be01721ee6#.ltv983lux Would those become part of the Resources page?

- **Discussion forums** is something that came up in our sessions at camp. Did you discuss implementing anything along these lines, or do you favor pushing all discussion to Slack?

- I don't see a **Camp** tab in your list. Are you planning to scrap that and only have it listed within the event listings?

**Comment by terichadbourne** [comment]
I just happened on the partially-written post about our website discussion from the Catskills, and thought it worth looking back on as we move forward with the redesign. Draft here: https://medium.com/offline-camp/building-a-central-community-hub-for-offline-first-7adb38f46d2c

A couple of the major points that came out of that Catskills discussion: 

_We envision the offlinefirst.org site as a central resource that can address three main goals:
1. Describing WHAT Offline First is and WHY it’s important
2. Exploring HOW one builds an effective Offline First app
3. Building the Offline First COMMUNITY_

I'm assuming that the community piece happens primarily in the _Contribute_ section you're proposing. Does the what/why live in the _Welcome_ and the _How_ live in a mix of the _Process_ and _Resources_?

Another thing we talked about was _identifying stakeholders and surrounding communities and making it easy for them to land on our site_. So, for example, there could be a landing page focused on IoT or one focused on Healthcare for people who are only getting to know Offline First through those entry points. Did you discuss anything like that in your recent chat, @gr2m @kidfribble?  One thing that might contribute to this goal is doing the faceted search engine (as suggested in my earlier comment) and having tags available related to industries. It would be nice to have more visually appealing landing pages, but worst case you could build the search engine so that a URL can take you to the search results for Healthcare tags, for example. And then in a Medium article about healthcare, we could easily link to all the relevant resources with a single reference.

**Comment by gr2m** [comment]
this is great input, thanks Teri!

**Comment by terichadbourne** [comment]
New additions since Offline Camp Berlin that will need homes on the updated website: 
- the sync matrix which there's a first draft of online already (thx @bradley-holt!), but not linked to yet from anywhere on the site (http://offlinefirst.org/sync/) - see #sync-matrix channel in Slack for discussion on this
- PeerDrop P2P example  (https://github.com/0x00A/peerdrop) - see discussion btw Fabian @filtercake & @janl in #camp Slack channel re where to locate this on the site.

**Comment by IrinaJay** [comment]
Thanks for the great input. Gregor and I had a few thoughts about it: 



> For the **Resources** page, does that have to be a static list where each resources sits under one category, and requires more and more scrolling as time goes on, or can it be a faceted search engine? Ideally we would have some database of all the resources we've found, tagged with format types (case study, event, video, podcast, interview, blog, article, etc.) and relevant content tags (UX, service workers, pouchdb, etc), and the user could control what subset of this they saw on the website. 

Ideally we have all data on GitHub so we can collaborate on it, If we’d have a separate database, we’d need a content management system to be able to collaborate with the community. Instead we suggest to use Jekyll posts for *Resources* similar to how we use posts for team members and sponsors today. The posts would have properties like topics (IoT, health, emerging markets, design etc.) medium (podcast, video, case study, blog etc.). Based on these properties we could implement a simple search/ filter for the *Resources* section.

We could also add a side wide search using Google custom search: https://cse.google.com/cse/  

> Re **News** and **Resources** - I would hope that all the awesome things @renrutnnej has found for the newsletter would also be included in the resources database and pull into listings of videos, case studies, tutorials, etc. there. In my mind, any suggestions people have for resources would go into one central spot to be added to both the site and the newsletter. Anything I could find (slowly) by opening each old newsletter and scrolling through I could also find instead by doing the faceted search in the resources database. (I was longing for this when I scoured the newsletter for podcasts recently - the tags in the newsletter were super helpful, but the process was time-consuming.)   We could definitely save overall time by making it easier for Jenn & I to share resources we're hunting down instead of both doing our own hunting (currently her for newsletter and me for camp Twitter and YouTube).

We are thinking about *News* being Jekyll posts as well, but in a short form. News are presented in short and are linked to the full article, blog, interview, podcasts etc, in the *Resources* section or anywhere on the web. The idea is not to create an additional blog, but rather link it to the original source. However, the news post would have a title and a short summary/ description of the content. 
E.g. if there is a new post on the offline camp blog, we could also post a short news post that links to the original blog in the *News* section.

> You included **Events** in your bulleted list but not your detailed page listing underneath. Do you still see that being its own page? If so, some  issues I'd like to account for there include: 
(1) Being able to list multiple sessions per event, with separate URLs for each session plus one for the overall event.
(2) Ability to differentiate btw event formats (conference, meetup, etc.)
(3) Ability to include speaker names and link to their Twitter accounts or whatnot
^^^ I can force this all to happen manually if needed, just suggesting we could build in styling for the options we're most likely to need
(4) Standard approach for dealing with past events and recorded sessions (right now we have a link to our YouTube channel
(5) Is there a case for storing events as resources as we do the others (see my database suggestion below). If so, would it be helpful to have records to which we later add links to slide decks and videos of completed presos?

We are thinking of having the *Events* listings more like the *News* section, where we would post a short description of the event (e.g. offline camp) and link to the event’s page. Meaning for the “official” events (offline camp, offline conference etc.) we would link to the dedicated website for these events (e.g. offlinecamp.org). Each event would have a date and location (country & city) that we could show on a map. 
Everyone in the community can suggest an event which they organize or participate in.

> Note that the website currently doesn't have a good place for **podcast** episodes, which I'm instead listing here: https://medium.com/offline-camp/offline-first-podcasts-d0be01721ee6#.ltv983lux Would those become part of the Resources page?

Yes

> **Discussion forums** is something that came up in our sessions at camp. Did you discuss implementing anything along these lines, or do you favor pushing all discussion to Slack?

We’d suggest to have all discussions in Slack. GitHub issues are not well suited for generic discussions. We could also suggest to post questions on Stack Overflow using the *offlinefirst* tag.

> I don't see a **Camp** tab in your list. Are you planning to scrap that and only have it listed within the event listings?

We would suggest to have dedicated website for offline camp and link to it from both the *Events* section as well as the *offlinefirst.org* landing page  

> Another thing we talked about was identifying stakeholders and surrounding communities and making it easy for them to land on our site. So, for example, there could be a landing page focused on IoT or one focused on Healthcare for people who are only getting to know Offline First through those entry points. Did you discuss anything like that in your recent chat, @gr2m @kidfribble? One thing that might contribute to this goal is doing the faceted search engine (as suggested in my earlier comment) and having tags available related to industries. It would be nice to have more visually appealing landing pages, but worst case you could build the search engine so that a URL can take you to the search results for Healthcare tags, for example. And then in a Medium article about healthcare, we could easily link to all the relevant resources with a single reference.

Topics can have their own landing page (e.g. offlinefirst.org/resources/health). This would also be really good for SEO.

**Comment by IrinaJay** [comment]
Gregor and I worked on the new website concept which you can find here: 
https://docs.google.com/a/thehoodiefirm.com/document/d/1gf7t7bvIE0rQ8WNR01RicRG4yv9zXDFDyjpwK7SEsVM/edit?usp=sharing

We tried to consider everything that's been discussed in this issue, please feel free to edit and comment the doc. Any feedback is highly appreciated.

**Comment by terichadbourne** [comment]
@IrinaJay Just a heads up that that Google Docs link is giving an access denied error, so I've requested permission. 

I'm just now seeing your responses to my questions from about a month ago, but I'll wait and take a look at that Google Doc before I respond. 

Thanks!!

**Comment by gr2m** [comment]
@terichadbourne can you try again?

**Comment by terichadbourne** [comment]
@gr2m Working now. Thank you!

**Comment by IrinaJay** [comment]
@terichadbourne THX for notifying me. The document should now be open for anyone who wants to contribute.

**Comment by terichadbourne** [comment]
@gr2m @IrinaJay I have a few questions/comments specifically in respect to your response to my questions above. Would it be helpful for me to put those here in GH or should I instead find places for them in the Google Doc?

**Comment by IrinaJay** [comment]
@terichadbourne I think GH would be great for comments to our responses.

**Comment by terichadbourne** [comment]
**@gr2m @IrinaJay My feedback on your feedback is in bold below. :)**

Thanks for the great input. Gregor and I had a few thoughts about it:

>For the Resources page, does that have to be a static list where each resources sits under one category, and requires more and more scrolling as time goes on, or can it be a faceted search engine? Ideally we would have some database of all the resources we've found, tagged with format types (case study, event, video, podcast, interview, blog, article, etc.) and relevant content tags (UX, service workers, pouchdb, etc), and the user could control what subset of this they saw on the website.

Ideally we have all data on GitHub so we can collaborate on it, If we’d have a separate database, we’d need a content management system to be able to collaborate with the community. Instead we suggest to use Jekyll posts for Resources similar to how we use posts for team members and sponsors today. The posts would have properties like topics (IoT, health, emerging markets, design etc.) medium (podcast, video, case study, blog etc.). Based on these properties we could implement a simple search/ filter for the Resources section.


We could also add a side wide search using Google custom search: https://cse.google.com/cse/


**Based on how Jekyll works, do you know if there can be multiple mediums or multiple topics per post? I can definitely see a common need for that.**

**Although I'm not opposed to using Jekyll posts and allowing contributions directly via GitHub, I'm actually quite concerned that there's no alternate plan for non-techy people who would find GH very overwhelming, creating either a real or perceived barrier to contributions that goes against our goal of inclusiveness of everyone - even non-developers.  This openness to contributions from non-developers is something we discussed and agreed to prioritize in both camp sessions, and I think it's particularly relevant since business-y people have a ton of content that we want more of, such as case studies.  I understand that your proposed process doesn't require someone to take a coding class, but from a purely emotional basis I can tell you that it is very overwhelming to a layperson and would require creating a GH account and referencing a lot of tutorial and web resources to perform what could be a simpler task. (Even as someone who already knew how to use the pencil button to edit files on GH, I had trouble figuring how to add a new .md file to do a similar process to submit to the What the Fest Berlin event list. )  I see this as an issue across all submission formats, from resources to news to event listings.**

**My proposed solution: If Jekyll posts are the preferred method for setting up contributions, and we're going to have to lay out a structure people use to make the .md files, then I'd love to see us invest the time to also make a GUI submission form accessible from the website that uses GitHub APIs to take its submissions and turn them into .md files submitted as issues. Someone who finds GH offputting could fill out this form which auto-generates a file and opens a pull request, while folks who love GH could use it to create their own .md file and open a pull request. In either case, the process is the same for people who are responsible for reviewing and accepting the request.   (I've discussed this briefly with a developer friend to confirm it's actually a thing that's possible, and possibly even something I could help figure out once I learn more about APIs.)**

>Re News and Resources - I would hope that all the awesome things @renrutnnej has found for the newsletter would also be included in the resources database and pull into listings of videos, case studies, tutorials, etc. there. In my mind, any suggestions people have for resources would go into one central spot to be added to both the site and the newsletter. Anything I could find (slowly) by opening each old newsletter and scrolling through I could also find instead by doing the faceted search in the resources database. (I was longing for this when I scoured the newsletter for podcasts recently - the tags in the newsletter were super helpful, but the process was time-consuming.) We could definitely save overall time by making it easier for Jenn & I to share resources we're hunting down instead of both doing our own hunting (currently her for newsletter and me for camp Twitter and YouTube).

We are thinking about News being Jekyll posts as well, but in a short form. News are presented in short and are linked to the full article, blog, interview, podcasts etc, in the Resources section or anywhere on the web. The idea is not to create an additional blog, but rather link it to the original source. However, the news post would have a title and a short summary/ description of the content.
E.g. if there is a new post on the offline camp blog, we could also post a short news post that links to the original blog in the News section.

**I think I'm having trouble understanding the difference between the resources page and the news page. Is there one database/post entry per resource, with a couple extra fields added if it needs to be surfaced on the news page as well? In my experience with the newsletter, almost everything there is something I'd want to find later in the resources section, and I think most of the resources we would add to a resource page, based on the newness of adding them, would make them candidates for the newsletter. What's the benefit of the two separate pages? Could we have the items from the most recent newsletter be marked with some kind of tag that puts them at the top of a resources page in a special "Latest" section (which is where someone would go to find the same things they could see in the newsletter email)? Please do push back on me here, as I may just not be understanding the distinction being proposed.**

**I'm totally good with having these be essentially links out to more detailed resources that live elsewhere in almost every case. There are a few community resources though that are being created directly on the website which I would suggest should also be surfaced here, like the sync matrix or PeerDrop example I mentioned in a previous comment, whose links could just go to a subpage of our own website if they do keep living on the site.**

>You included Events in your bulleted list but not your detailed page listing underneath. Do you still see that being its own page? If so, some issues I'd like to account for there include:
(1) Being able to list multiple sessions per event, with separate URLs for each session plus one for the overall event.
(2) Ability to differentiate btw event formats (conference, meetup, etc.)
(3) Ability to include speaker names and link to their Twitter accounts or whatnot
^^^ I can force this all to happen manually if needed, just suggesting we could build in styling for the options we're most likely to need
(4) Standard approach for dealing with past events and recorded sessions (right now we have a link to our YouTube channel
(5) Is there a case for storing events as resources as we do the others (see my database suggestion below). If so, would it be helpful to have records to which we later add links to slide decks and videos of completed presos?

We are thinking of having the Events listings more like the News section, where we would post a short description of the event (e.g. offline camp) and link to the event’s page. Meaning for the “official” events (offline camp, offline conference etc.) we would link to the dedicated website for these events (e.g. offlinecamp.org). Each event would have a date and location (country & city) that we could show on a map.
Everyone in the community can suggest an event which they organize or participate in.

**The plan of only having four characteristics per event (name, URL, location, dates) doesn't fit with the way we're currently using the page. It would, in my opinion, limit us to listing events that are essentially all about Offline First, and at present those are only the events that we create ourselves, only a few a year, which leads to a pretty meager listing. If you check the page today, you'll see that I recently updated it with lots of upcoming content, and I use it to show events that include some Offline First content - highlighting the contributions of community members who got a session accepted at a broader show or meetup. That means that we need to display for each full event the event name, location, dates, URL, and format (conference, meetup, webinar, etc.) and also be able to have one or more session listings attached which each have a session title, URL to the abstract, speaker name, speaker twitter handle.  (I've been meaning to talk to @kidfribble about how to get more flexibility with formatting to make some of this easier on the existing site.) Given that one of our most promising means of spreading the good word about Offline First is to embed our content in a variety of shows with different focuses, and given that companies supporting the movement need evidence that the message is spreading, a list like this feels much more appropriate to me than a list focused on the few shows fully dedicated to the subject. It will also make the great map idea look more exciting w/ more pins!**

**As mentioned above, I would like to see us continue to use the events page as a place to continue to surface a link to the past presentation videos shared in our YouTube channels. It could also be a place to encourage people to submit slideshares or recorded talk videos as resource submissions (and I could use those video submissions to find things to add to the YouTube channel).**

**Would these same event listing post files be surfaceable in the news or resources sites? As long as the date is in the future, they seem relevant for the newsletter, while perhaps only related videos or slide decks would be relevant for events that occurred in the past.**

>Note that the website currently doesn't have a good place for podcast episodes, which I'm instead listing here: https://medium.com/offline-camp/offline-first-podcasts-d0be01721ee6#.ltv983lux Would those become part of the Resources page?

Yes

**Great.**

>Discussion forums is something that came up in our sessions at camp. Did you discuss implementing anything along these lines, or do you favor pushing all discussion to Slack?

We’d suggest to have all discussions in Slack. GitHub issues are not well suited for generic discussions. We could also suggest to post questions on Stack Overflow using the offlinefirst tag.

**I agree that Slack is better for discussion, but I do have a major concern in that our free Slack account is completely full right now, so every new message is pushing an old one out of the history. Getting reliable corporate sponsorship to upgrade to a paid account would make this much more viable as a place to really have the history of our discussion.**

**Regarding the Stack Overflow suggestion, maybe we could include a link that takes you to a filtered list of all open SO questions on Offline First so you can try to help answer them?**

>I don't see a Camp tab in your list. Are you planning to scrap that and only have it listed within the event listings?

We would suggest to have dedicated website for offline camp and link to it from both the Events section as well as the offlinefirst.org landing page

**I'm not someone who would understand the technical or SEO or hosting implications of this, but I don't have any objections. :)**

>Another thing we talked about was identifying stakeholders and surrounding communities and making it easy for them to land on our site. So, for example, there could be a landing page focused on IoT or one focused on Healthcare for people who are only getting to know Offline First through those entry points. Did you discuss anything like that in your recent chat, @gr2m @kidfribble? One thing that might contribute to this goal is doing the faceted search engine (as suggested in my earlier comment) and having tags available related to industries. It would be nice to have more visually appealing landing pages, but worst case you could build the search engine so that a URL can take you to the search results for Healthcare tags, for example. And then in a Medium article about healthcare, we could easily link to all the relevant resources with a single reference.

Topics can have their own landing page (e.g. offlinefirst.org/resources/health). This would also be really good for SEO.

**Awesome. Have you thought about the content layout of those sites? So for example, we could have a paragraph or two connecting the dots on why OF matters in healthcare, and then some handpicked favorite resources linked (such as the most compelling case studies or podcast episodes, culled from the resources database), and then a filtered list all the relevant resources we know exist.  I think it will be really important for us to build out a little bit of messaging for these rather then simply having them present a filtered version of the resources page.**

**Comment by IrinaJay** [comment]
I hope I covered all of the great input @terichadbourne

> Based on how Jekyll works, do you know if there can be multiple mediums or multiple topics per post? I can definitely see a common need for that.

yes.

> Although I'm not opposed to using Jekyll posts and allowing contributions directly via GitHub, I'm actually quite concerned that there's no alternate plan for non-techy people who would find GH very overwhelming, creating either a real or perceived barrier to contributions that goes against our goal of inclusiveness of everyone - even non-developers

Something we could do is keeping the Offline Camp medium. Most of the posts on offlinefirst.org will be very short and then link to the news on another page or a resource on offlinefirst.org. There are ready-made solutions for posting e.g. http://prose.io/ or we can build something custom ourselves. With prose.io you still need a GH account for it, but you don’t need any developer experience to sign up for GitHub. And there is no plan for comments on offlinefirst.org, that’s a feature, not a bug. We could still add a 3rd party service like https://disqus.com/, but we probably won’t need to.

The benefit of having the content part of the repository is that it’s easier make it work offline

> I think I'm having trouble understanding the difference between the resources page and the news page

News and Resources belong together in a way that News is the latest piece of information that’s recently been posted on the website. It has a chronological order and each post is an abstract leading to the original source on resources or a location outside of offlinefirst.org.

>  Is there one database/post entry per resource, with a couple extra fields added if it needs to be surfaced on the news page as well? (…) almost everything there is something I'd want to find later in the resources section (…) What's the benefit of the two separate pages? 

At least in the news section, yes. But for example for Progressive Web Apps there is so much content that it’s overwhelming, and we can help by maintaining a useful list of resources about them, instead of keeping everything around forever. The benefit would be for people to look at news knowing what’s recently been added and as well as seeing all the info the (latest) Newsletter contains while Resources is a library of all the knowledge shared so far.


> Could we have the items from the most recent newsletter be marked with some kind of tag that puts them at the top of a resources page in a special "Latest" section

We can most certainly add a “latest” tag (teri the Latest section basically is the News section, since the things from the Newsletter will go there as well). 

> There are a few community resources though that are being created directly on the website which I would suggest should also be surfaced here, like the sync matrix or PeerDrop example I mentioned in a previous comment, whose links could just go to a subpage of our own website if they do keep living on the site.

Yeah that’s no problem

> The plan of only having four characteristics per event (name, URL, location, dates) doesn't fit with the way we're currently using the page

I don’t mind the format at all, I agree that we should be able to list a wide range of different kind of events and ask people to contribute their own. We can also have different types of events.

> Slack / forums

¯\_(ツ)_/¯

The challenge is that most people are volunteers that don’t have time to be around all the time, it’s more like a place for the community to hang out and chat. If chats disappear over time, that should be no problem. For content that is more important we should put it on the website, if someone submits e.g. a resource there can still be discussion about it in the pull request

> Have you thought about the content layout of those sites

No not yet, that’s the next step. But we can certainly mix dynamic, generated content with curated introductions per landing page

**Comment by terichadbourne** [comment]
Thanks @IrinaJay!

> Something we could do is keeping the Offline Camp medium. Most of the posts on offlinefirst.org will be very short and then link to the news on another page or a resource on offlinefirst.org. There are ready-made solutions for posting e.g. http://prose.io/ or we can build something custom ourselves. With prose.io you still need a GH account for it, but you don’t need any developer experience to sign up for GitHub. And there is no plan for comments on offlinefirst.org, that’s a feature, not a bug. We could still add a 3rd party service like https://disqus.com/, but we probably won’t need to.

My comment relating to allowing non-techy people to contribute via a GUI versus GH was specifically meant in reference to how they could contribute resources, news, and events that they thought should show up on offlinefirst.org (which would otherwise be created by adding an .md file in GH.   Although I was definitely anticipating the Offline Camp Medium continuing to exist (some of the articles there could be captured / linked out to in "resources" on the new website), it doesn't solve that particular problem. I've only glanced briefly at it, but Prose.io might do what I'm suggesting, or we could build our own user-friendly form somehow if not.

I didn't mean to be recommending a comment feature for the website.

> The challenge is that most people are volunteers that don’t have time to be around all the time, it’s more like a place for the community to hang out and chat. If chats disappear over time, that should be no problem. For content that is more important we should put it on the website, if someone submits e.g. a resource there can still be discussion about it in the pull request

Regarding whether we need a place for the historical conversation that's disappearing in Slack, you're totally right that if there's an attachment or link to a resource we love shared on Slack that we should be adding it as a resource on the website. (For example, the great design poster Jonah made that has disappeared in Slack history.)  Part of me wishes there were a way for people to see past conversations, because we often get newcomers to Slack asking the same questions as previous newcomers and those who are there more regularly answering those questions repeatedly in similar ways. To make Slack work to solve that problem it would probably need to both be paid by a generous sponsor and have someone encouraging more use of channels to focus on subtopics. I think I remember someone at a camp mentioning Disqus as a more organized option if we wanted something more forum-y, but I'm not specifically advocating for that right now or suggesting it would need to live on the website.  

I don't personally view PRs or issues on GH as great spots for community discussion since they suppose some familiarity with GH that might keep less techy folks from contributing. I'd like for us to build the website in a way where Joe Schmo can come and view it like they would any other website, not having to understand the inner workings or find their way backstage to locate a hidden feature. Making a link to Slack or another user-friendly chat option obvious on the website, even though that chat platform would live outside the website, would be preferable in my view.

I think I'm still not 100% following you on the various aspects of the news v latest resources distinction, but I'll probably have it all straight as soon as I see a mock-up. It's quite possible we're saying the same thing in different ways. :)

>But for example for Progressive Web Apps there is so much content that it’s overwhelming, and we can help by maintaining a useful list of resources about them, instead of keeping everything around forever.

If I understand you correctly, I'm not sure I agree that there's a reason to remove resources from the site, unless someone lets us know that a specific resource is no longer accurate or in existence. (I loved your earlier suggestion to allow folks to come to us and call those cases out.) But of course they wouldn't stay listed as "news" or "latest" for long, I'm just proposing they stay in the searchable / sortable "resources" section.

>I don’t mind the format at all, I agree that we should be able to list a wide range of different kind of events and ask people to contribute their own. We can also have different types of events.

Glad to hear we can work on a more flexible formatting system for event listings. I'm happy to help with structuring that when the time comes. 

Would you like me to move any of my notes over to the Google Doc for you? What are the next steps and how can I help?

**Comment by IrinaJay** [comment]
Thx @terichadbourne :)

> Part of me wishes there were a way for people to see past conversations, because we often get newcomers to Slack asking the same questions as previous newcomers and those who are there more regularly answering those questions repeatedly in similar ways. To make Slack work to solve that problem it would probably need to both be paid by a generous sponsor and have someone encouraging more use of channels to focus on subtopics. I think I remember someone at a camp mentioning Disqus as a more organized option if we wanted something more forum-y, but I'm not specifically advocating for that right now or suggesting it would need to live on the website.

Would it be helpful to make an FAQ kind of place to walk new people through the community and the frequently asked questions? The downside with this would be that someone would have to go through the questions (which are probably already overwhelming) and make the FAQ as well as keep it updated. But I think it might be a great starting point though for those just starting out with offline first.

> I don't personally view PRs or issues on GH as great spots for community discussion since they suppose some familiarity with GH that might keep less techy folks from contributing. I'd like for us to build the website in a way where Joe Schmo can come and view it like they would any other website, not having to understand the inner workings or find their way backstage to locate a hidden feature. Making a link to Slack or another user-friendly chat option obvious on the website, even though that chat platform would live outside the website, would be preferable in my view.

Completely agree.

> I think I'm still not 100% following you on the various aspects of the news v latest resources distinction, but I'll probably have it all straight as soon as I see a mock-up. It's quite possible we're saying the same thing in different ways. :)

I have the feeling we are talking about the same thing just different :D

> If I understand you correctly, I'm not sure I agree that there's a reason to remove resources from the site, unless someone lets us know that a specific resource is no longer accurate or in existence. (I loved your earlier suggestion to allow folks to come to us and call those cases out.) But of course they wouldn't stay listed as "news" or "latest" for long, I'm just proposing they stay in the searchable / sortable "resources" section.

I think I might have misphrased that. The content will definitely stay in the resources section and be available for the search.  We just need to find a way to structure older content in a way it doesn’t overwhelm people when looking at the resources section. And we thought it might be good to have some kind of archive sorted by year and month. What do you think?

>Would you like me to move any of my notes over to the Google Doc for you? What are the next steps and how can I help?

I believe @gr2m has some ideas on what we could do next.

**Comment by gr2m** [comment]
Just to be clear: people _will_ use a web UI to submit news, resources or events, they _will not_ need to use GitHub’s UI for it. They will only need a GitHub account and we will add a "Login with your GitHub Account" feature. Once logged in, people will see the additional UIs to submit new content. The UI will directly talk to the GitHub API to create the markdown files (or what ever is needed) in the background.

Maybe the next step is to show what I mean by all that, with a simplified prototype?

**Comment by terichadbourne** [comment]
@IrinaJay 

>Would it be helpful to make an FAQ kinda of place to walk new people through the community and the frequently asked questions? The downside with this would be that someone would have to go through the questions (which are probably already overwhelming) and make the FAQ as well as keep it updated. But I think it might be a great starting point though for those just starting out with offline first.

I agree that this would be a ton of work to maintain. The kinds of questions I see coming up most by newbies on Slack are things like "should I use Couch & Pouch," to which everyone has to ask more questions about the person's use case before answering. So if there were a well-used Couch/Pouch channel with history, the new guest could read back through and glean some insight before asking a more specific question. I'm not sure that case would translate particularly well to an FAQ. 

I wonder if there would be a use case for a "Getting started" tag that could be applied to certain resources that are particularly beginner-level, so someone could go to resources and search for "CouchDB," "PouchDB," and "Getting Started" and see what pops up. 

When I (with few tech skills) see newbies asking questions in Slack, what I often do is point them to the resources page on Medium (which shows where to find the Medium pub, YouTube channel, podcast listings, etc.). I also sometimes point them to our Medium pub with search results built into the URL, to get them to all the Service Worker articles, for example. So even just the sheer fact of having a well-organized resources section will solve some of these issues.  I wonder if we could program a bot to greet all new members of the Slack team with a friendly message that points them to the resources page of the website. (Though we of course want humans welcoming them as well.)

There might be a case to be made for using an FAQ to address things like "Why Offline **First**, why not **also** or eventually?" or "What's the difference between Offline First and Progressive Web Apps?" - stuff that might be hard to get at in concise overarching messaging but which deserves a closer look.

>I think I might have misphrased that. The content will definitely stay in the resources section and be available for the search. We just need to find a way to structure older content in a way it doesn’t overwhelm people when looking at the resources section. And we thought it might be good to have some kind of archive sorted by year and month. What do you think?

Glad to hear we're on the same page about keeping resources available. When I think of how I would use a resources page, I think it would almost always be by searching for a tag or combo of tags (use case, PouchDB, Service Worker, podcast, video, etc.) and wanting to see everything that comes up for that. The only possible reason I could think of that I'd want to search based on date of posting would be if I were thinking "Oh, crap, I remember seeing something cool in the newsletter last month but I don't remember the details." Overall, I don't see a strong reason for the dated archive, but having things searchable by date just in case would cover us.

@gr2m 
>Just to be clear: people will use a web UI to submit news, resources or events, they will not need to use GitHub’s UI for it. They will only need a GitHub account and we will add a "Login with your GitHub Account" feature. Once logged in, people will see the additional UIs to submit new content. The UI will directly talk to the GitHub API to create the markdown files (or what ever is needed) in the background.

Really glad to hear that you're already planning to have people use a friendly UI that doesn't require knowing code or markdown! 

> Maybe the next step is to show what I mean by all that, with a simplified prototype?

Sounds great!

**Comment by gr2m** [comment]
@terichadbourne @IrinaJay here is a screencast showing how we can implement a login on offlinefirst.org, let people submit links (or in our case news, resources, events ... everything we want) and also how people with write access to the repository can review these submissions directly from the UI. At no point does anyone need to use GitHub directly, but we can still use features like automated tests that check if the URL works or if it’s a duplicate for example, in which case we can block the PR from being merged

https://dl.dropboxusercontent.com/u/732913/offlinefirst/submitting-and-accepting-pull-requests-through-web-ui.mov

Hope that gives you a good idea of how we can make the experience of contributing to the website very nice while still benefiting from the tooling that GitHub gives us

**Comment by terichadbourne** [comment]
This is super helpful, @gr2m. Thanks for taking the time to pull that together! That definitely addresses my concern of making it easy for non-coders to make contributions.
